### DISCUSSION TITLE
Setting Up Arduino for SPI Communication with a 24-Channel LED Driver

### DISCUSSION THREAD

**Discussion by haeinkim** [bonsai-rx/bonsai#2039] [Q&A]
Hello,

I am trying to use this 24-channel LED driver (https://www.adafruit.com/product/1429) to control multiple LEDs individually in a closed-loop manner. This driver requires three SPI pins for communication, sending 3-5V logic SPI data (12-bit PWM output) on DIN (data in), CLK (clock), and LAT (latch). How do I set up Arduino for this specific communication? With my limited knowledge, I believe the standard Firmata cannot handle this type of communication.

Thanks in advance for your help!

**Comment by bruno-f-cruz** [comment]
I think you will need to use a modified version of firmata. I have not attempted it before, but there seems to exist documentation in their official repository: https://github.com/firmata/protocol/blob/master/spi.md
